### DISCUSSION TITLE
Error Code: Missing offer | Add missing offer details

### DISCUSSION THREAD

**Discussion by cnr91** [amzn/selling-partner-api-models#4163] [Q&A] [answered]
When I use Bulk Listing, the product is added to my inventory, but I get a warning from amazon that I don't understand?
I have tried all possibilities, but I can't solve it, can you help me?

Amazon Inventory:
![image](https://github.com/user-attachments/assets/4ae3bb07-9588-4f48-bc5b-a34761bebe23)

API TOOL:
![image](https://github.com/user-attachments/assets/3788ecd1-aa9c-474c-a31b-b266e2f9da9a)

```
{
    "header": {
        "sellerId": "AB*************",
        "version": "2.0",
        "issueLocale": "en_US"
    },
    "messages": [
        {
            "messageId": 1,
            "sku": "20-0000-0001",
            "operationType": "UPDATE",
            "productType": "SKIN_MOISTURIZER",
            "requirements": "LISTING_OFFER_ONLY",
            "attributes": {
                "merchant_suggested_asin": [
                    {
                        "value": "B06XHPB9SG",
                        "marketplace_id": "A39IBJ37TRP1C6"
                    }
                ],
                "fulfillment_availability": [
                    {
                        "fulfillment_channel_code": "DEFAULT",
                        "quantity": 3,
                        "lead_time_to_ship_max_days": 5
                    }
                ],
                "condition_type": [
                    {
                        "value": "new_new",
                        "marketplace_id": "A39IBJ37TRP1C6"
                    }
                ],
                "merchant_shipping_group": [
                    {
                        "value": "legacy-template-id",
                        "marketplace_id": "A39IBJ37TRP1C6"
                    }
                ],
                "purchasable_offer": [
                    {
                        "currency": "AUD",
                        "marketplace_id": "A39IBJ37TRP1C6",
                        "our_price": [
                            {
                                "schedule": [
                                    {
                                        "value_with_tax": 69.9
                                    }
                                ]
                            }
                        ],
                        "minimum_seller_allowed_price": [
                            {
                                "schedule": [
                                    {
                                        "value_with_tax": 59.9
                                    }
                                ]
                            }
                        ],
                        "maximum_seller_allowed_price": [
                            {
                                "schedule": [
                                    {
                                        "value_with_tax": 79.9
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        }
    ]
}
```

**Comment by mafge** [comment]
Hi @cnr91,
thanks for reaching out!
As far as I can see you didn't specify the pricing details. Without this information Amazon can't display your listing on the website. Hope that helps!
Best,
Marc
Selling Partner Developer Services

**Comment by cnr91** [comment]
> Hi @cnr91,
> 
> thanks for reaching out!
> 
> As far as I can see you didn't specify the pricing details. Without this information Amazon can't display your listing on the website. Hope that helps!
> 
> Best,
> 
> Marc
> 
> Selling Partner Developer Services

I don't understand exactly what you means. I added our price, min price and maximum price in json. I wonder what field I'm forgetting

By the way, after 1.5 hour the missing offer error disappeared from the view, but it took a long time to recover

**Comment by mafge** [comment]
@cnr91

> By the way, after 1.5 hour the missing offer error disappeared from the view, but it took a long time to recover

Just to make sure that I get that right: Was the price applied now?

I am asking because (depending on the size of your feed) it can sometimes take some time for the backend until all listings are fully processed. In order to optimize feed processing times, please make sure to follow our [best practices](https://developer-docs.amazon.com/sp-api/docs/feeds-api-best-practices).

**Reply by cnr91** [reply]
Yes, the pricing information with the JSON I sent above was fixed after 1.5 hours. It is not nice that it is so late because I may need to update the prices every hour. This was not a problem when updating prices with TSV.

**Reply by mafge** [reply, marked as answer]
In your case it might make sense to use patchListingsItem from Listings Items API instead of using the feed (especially when the number of items is < 1.500).
